modules/kernel/src/main/scala/tofu/Clock.scala

```diff
@@ -0,0 +1,67 @@
+package tofu
+
+import tofu.internal.EffectComp
+import tofu.internal.carriers._
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.FiniteDuration
+
+trait Clock[F[_]] {
+
+  /** Returns the current time, as a Unix timestamp (number of time units
+    * since the Unix epoch), suspended in `F[_]`.
+    *
+    * This is the pure equivalent of Java's `System.currentTimeMillis`,
+    * or of `CLOCK_REALTIME` from Linux's `clock_gettime()`.
+    */
+  def realTime(unit: TimeUnit): F[Long]
+
+  /** Returns a monotonic clock measurement, if supported by the
+    * underlying platform.
+    *
+    * This is the pure equivalent of Java's `System.nanoTime`,
+    * or of `CLOCK_MONOTONIC` from Linux's `clock_gettime()`.
+    */
+  def nanos: F[Long]
+}
+
+object Clock extends EffectComp[Clock] with ClockInterop
+
+trait Sleep[F[_]] {
```

Comment:
This one is more like Fiber-concurrency-related stuff, so I think it should have its own file

Reply:
No, it's unrelated to Fiber and concurrency, it doesn't imply some branching of execution, so I disagree here